Panic when manifest has no version stanzas

During testing I commented out all versions from the `goreleaser` package and ran `hermit install goreleaser` and got the following panic:

```
panic: runtime error: index out of range [-1] [recovered]
	panic: runtime error: index out of range [-1] [recovered]
	panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/cashapp/hermit/app.Main.func3()
	/home/runner/work/hermit/hermit/app/main.go:143 +0x59
panic({0x151ae00, 0xc0000e2c00})
	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/panic.go:1038 +0x215
github.com/alecthomas/kong.catch(0xc0001937c8)
	/home/runner/go/pkg/mod/github.com/alecthomas/kong@v0.2.18-0.20210713035501-d1a818b5a1a8/kong.go:407 +0xa5
panic({0x151ae00, 0xc0000e2c00})
	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/panic.go:1038 +0x215
github.com/cashapp/hermit/manifest.newPackage(0xc00030d8b0, {{0xc0000988d0, 0x26}, {0xc000240090, 0x24}, {0x154925a, 0x6}, {0x15481f5, 0x5}}, {0x1602d28, ...})
	/home/runner/work/hermit/hermit/manifest/resolver.go:349 +0x35eb
github.com/cashapp/hermit/manifest.(*Resolver).Resolve(0xc0002d6c80, 0x0, {0x1602d28, 0xc00009adc0})
	/home/runner/work/hermit/hermit/manifest/resolver.go:292 +0x185
github.com/cashapp/hermit.(*Env).Resolve(0xc00009adc0, 0xc00028e39c, {0x1602d28, 0xc00009adc0}, 0x0)
	/home/runner/work/hermit/hermit/env.go:586 +0xd8
github.com/cashapp/hermit.(*Env).ResolveWithDeps(0xc00028e39c, 0xe, {0xc0000a2618, 0x1, 0x1}, {0x1602d28, 0xc00009adc0}, 0x0)
	/home/runner/work/hermit/hermit/env.go:1061 +0x2b7
github.com/cashapp/hermit/app.(*installCmd).Run(0xc0000b6180, 0xc00021e180, 0x1, 0x1)
	/home/runner/work/hermit/hermit/app/install_cmd.go:69 +0x5eb
reflect.Value.call({0x14cd4c0, 0xc0000b6180, 0x60}, {0x1547580, 0x4}, {0xc0002d3200, 0x3, 0x30})
	/opt/hostedtoolcache/go/1.17.1/x64/src/reflect/value.go:543 +0x814
reflect.Value.Call({0x14cd4c0, 0xc0000b6180, 0x9}, {0xc0002d3200, 0x3, 0x4})
	/opt/hostedtoolcache/go/1.17.1/x64/src/reflect/value.go:339 +0xc5
github.com/alecthomas/kong.callMethod({0x1546f6e, 0x3}, {0x14dbb40, 0xc0000b6180, 0x3}, {0x14cd4c0, 0xc0000b6180, 0x100000001439e25}, 0x0)
	/home/runner/go/pkg/mod/github.com/alecthomas/kong@v0.2.18-0.20210713035501-d1a818b5a1a8/callbacks.go:71 +0x4e7
github.com/alecthomas/kong.(*Context).RunNode(0xc0000a0300, 0xc0000c1420, {0xc000193bb8, 0x7, 0x7})
	/home/runner/go/pkg/mod/github.com/alecthomas/kong@v0.2.18-0.20210713035501-d1a818b5a1a8/context.go:725 +0x40b
github.com/alecthomas/kong.(*Context).Run(0x152d6a0, {0xc000193bb8, 0x7, 0x7})
	/home/runner/go/pkg/mod/github.com/alecthomas/kong@v0.2.18-0.20210713035501-d1a818b5a1a8/context.go:742 +0xa5
github.com/cashapp/hermit/app.Main({{0xc0002149d0, 0x10}, 0x2, {0xc00021c340, 0x3a}, {0x1931960, 0xe, 0xe}, 0x1573d58, {{0x0, ...}, ...}, ...})
	/home/runner/work/hermit/hermit/app/main.go:249 +0x1a8f
main.main()
	/home/runner/work/hermit/hermit/cmd/hermit/main.go:34 +0x2b8
```

P.S. Looks like it might be worth building with `-trimpaths`?

--- a/manifest/resolver.go
+++ b/manifest/resolver.go
@@ -344,6 +344,9 @@ func newPackage(manifest *AnnotatedManifest, config Config, selector Selector) (
 		channelRefs, foundUpdateInterval, found = matchChannel(manifest, selector)
 		allRefs = append(allRefs, channelRefs...)
 	}
+	if len(allRefs) == 0 {
+		return nil, errors.Errorf("could not find any versions matching %s", selector)
+	}
 	// Finally just pick the most recent version.
 	if !found.IsSet() && !selector.IsFullyQualified() {
 		sort.Sort(allRefs)

